# Loan Status prediction

In [81]:
import pandas as pd

In [82]:
dataset=pd.read_csv("C://Users//user//Downloads//loan_prediction.csv")

##  Displaying Top 5 Rows of The Dataset

In [83]:
dataset.head()

Loan_ID Gender Married Dependents     Education Self_Employed  \
0  LP001002   Male      No          0      Graduate            No   
1  LP001003   Male     Yes          1      Graduate            No   
2  LP001005   Male     Yes          0      Graduate           Yes   
3  LP001006   Male     Yes          0  Not Graduate            No   
4  LP001008   Male      No          0      Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5849                0.0         NaN             360.0   
1             4583             1508.0       128.0             360.0   
2             3000                0.0        66.0             360.0   
3             2583             2358.0       120.0             360.0   
4             6000                0.0       141.0             360.0   

   Credit_History Property_Area Loan_Status  
0             1.0         Urban           Y  
1             1.0         Rural           N  
2             1.0         Urban           Y  
3             1.0         Urban           Y  
4             1.0         Urban           Y

## Exploring the dataset

In [84]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


In [85]:
dataset.shape

(614, 13)

In [86]:
dataset.isnull().sum()

Loan_ID               0
Gender               13
Married               3
Dependents           15
Education             0
Self_Employed        32
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount           22
Loan_Amount_Term     14
Credit_History       50
Property_Area         0
Loan_Status           0
dtype: int64

In [87]:
dataset.isnull().sum()*100/len(dataset)

Loan_ID              0.000000
Gender               2.117264
Married              0.488599
Dependents           2.442997
Education            0.000000
Self_Employed        5.211726
ApplicantIncome      0.000000
CoapplicantIncome    0.000000
LoanAmount           3.583062
Loan_Amount_Term     2.280130
Credit_History       8.143322
Property_Area        0.000000
Loan_Status          0.000000
dtype: float64

In [88]:
dataset.describe()

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count       614.000000         614.000000  592.000000         600.00000   
mean       5403.459283        1621.245798  146.412162         342.00000   
std        6109.041673        2926.248369   85.587325          65.12041   
min         150.000000           0.000000    9.000000          12.00000   
25%        2877.500000           0.000000  100.000000         360.00000   
50%        3812.500000        1188.500000  128.000000         360.00000   
75%        5795.000000        2297.250000  168.000000         360.00000   
max       81000.000000       41667.000000  700.000000         480.00000   

       Credit_History  
count      564.000000  
mean         0.842199  
std          0.364878  
min          0.000000  
25%          1.000000  
50%          1.000000  
75%          1.000000  
max          1.000000

In [89]:
dataset['Gender'].unique()

array(['Male', 'Female', nan], dtype=object)

In [90]:
dataset['Married'].unique()

array(['No', 'Yes', nan], dtype=object)

In [91]:
dataset['Dependents'].unique()

array(['0', '1', '2', '3+', nan], dtype=object)

In [92]:
#handling columns with less than 5% of null values and categorical in data

columns=['Gender','Married','Dependents']

dataset=dataset.dropna(subset=columns)

In [93]:
dataset.isnull().sum()*100/len(dataset)

Loan_ID              0.000000
Gender               0.000000
Married              0.000000
Dependents           0.000000
Education            0.000000
Self_Employed        5.460751
ApplicantIncome      0.000000
CoapplicantIncome    0.000000
LoanAmount           3.412969
Loan_Amount_Term     2.218430
Credit_History       8.361775
Property_Area        0.000000
Loan_Status          0.000000
dtype: float64

In [94]:
dataset['Self_Employed'].unique()

array(['No', 'Yes', nan], dtype=object)

In [95]:
dataset['Self_Employed'].mode()[0]

'No'

In [96]:
dataset['Self_Employed'] = dataset['Self_Employed'].fillna(dataset['Self_Employed'].mode()[0])

In [97]:
dataset.isnull().sum()*100/len(dataset)

Loan_ID              0.000000
Gender               0.000000
Married              0.000000
Dependents           0.000000
Education            0.000000
Self_Employed        0.000000
ApplicantIncome      0.000000
CoapplicantIncome    0.000000
LoanAmount           3.412969
Loan_Amount_Term     2.218430
Credit_History       8.361775
Property_Area        0.000000
Loan_Status          0.000000
dtype: float64

In [98]:
dataset['Self_Employed'].unique()

array(['No', 'Yes'], dtype=object)

In [99]:
#handling columns with less than 5% of null values and numerical in data
dataset['LoanAmount'].unique()


array([ nan, 128.,  66., 120., 141., 267.,  95., 158., 168., 349.,  70.,
       109., 200., 114.,  17., 125., 100.,  76., 133., 115., 104., 315.,
       116., 151., 191., 122., 110.,  35., 201.,  74., 106., 320., 144.,
       184.,  80.,  47.,  75., 134.,  96.,  88.,  44., 112., 286.,  97.,
       135., 180.,  99., 165., 258., 126., 312., 136., 172.,  81., 187.,
       113., 176., 130., 111., 167., 265.,  50., 210., 175., 131., 188.,
        25., 137., 225., 216.,  94., 139., 152., 118., 185., 154.,  85.,
       259., 194.,  93., 160., 182., 650., 102., 290.,  84., 242., 129.,
        30., 244., 600., 255.,  98., 275., 121.,  63.,  87., 101., 495.,
        67.,  73., 260., 108.,  58.,  48., 164., 170.,  83.,  90., 166.,
       124.,  55.,  59., 127., 214., 240.,  72.,  60.,  42., 280., 140.,
       155., 123., 138., 279., 192., 304., 330., 150., 207., 436.,  78.,
        54.,  89., 132., 480.,  56., 300., 376., 117., 490.,  71., 173.,
        46., 228., 308., 105., 236., 570., 380., 29

In [100]:
dataset['LoanAmount'].mode()[0]

120.0

In [101]:
dataset['LoanAmount']= dataset['LoanAmount'].fillna(dataset['LoanAmount'].mode()[0])

In [102]:
dataset.isnull().sum()*100/len(dataset)

Loan_ID              0.000000
Gender               0.000000
Married              0.000000
Dependents           0.000000
Education            0.000000
Self_Employed        0.000000
ApplicantIncome      0.000000
CoapplicantIncome    0.000000
LoanAmount           0.000000
Loan_Amount_Term     2.218430
Credit_History       8.361775
Property_Area        0.000000
Loan_Status          0.000000
dtype: float64

In [103]:
dataset['Loan_Amount_Term'].unique()

array([360., 120., 240.,  nan, 180.,  60., 300., 480.,  36.,  84.,  12.])

In [104]:
dataset['Loan_Amount_Term'].mode()[0]

360.0

In [105]:
dataset['Loan_Amount_Term']= dataset['Loan_Amount_Term'].fillna(dataset['Loan_Amount_Term'].mode()[0])

In [106]:
dataset['Credit_History'].unique()

array([ 1.,  0., nan])

In [107]:
dataset['Credit_History']= dataset['Credit_History'].fillna(dataset['Credit_History'].mode()[0])

In [108]:
dataset=dataset.drop('Loan_ID', axis=True)

In [109]:
dataset.isnull().sum()*100/len(dataset)

Gender               0.0
Married              0.0
Dependents           0.0
Education            0.0
Self_Employed        0.0
ApplicantIncome      0.0
CoapplicantIncome    0.0
LoanAmount           0.0
Loan_Amount_Term     0.0
Credit_History       0.0
Property_Area        0.0
Loan_Status          0.0
dtype: float64

## Handling the categorical data to improve readability

In [110]:
dataset.sample(5)

Gender Married Dependents     Education Self_Employed  ApplicantIncome  \
377   Male     Yes          0      Graduate            No             4310   
381   Male      No          0      Graduate            No             5941   
479   Male     Yes          2      Graduate            No             2947   
533   Male      No          1      Graduate            No            11250   
322   Male     Yes          2  Not Graduate            No             3601   

     CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
377                0.0       130.0             360.0             1.0   
381             4232.0       296.0             360.0             1.0   
479             1603.0       120.0             360.0             1.0   
533                0.0       196.0             360.0             1.0   
322             1590.0       120.0             360.0             1.0   

    Property_Area Loan_Status  
377     Semiurban           Y  
381     Semiurban           Y  
479         Urban           N  
533     Semiurban           N  
322         Rural           Y

In [130]:
dataset['Property_Area'].unique()

array(['Urban', 'Rural', 'Semiurban'], dtype=object)

In [112]:
dataset['Gender'] = dataset['Gender'].map({'Male':1,'Female':0}).astype('int')

In [113]:
dataset['Married']= dataset['Married'].map({'Yes' :1,'No':0}).astype('int')

In [114]:
dataset['Education'] =dataset['Education'].map({'Graduate' :1, 'Not Graduate' :0}).astype('int')

In [115]:
dataset['Property_Area'] =dataset['Property_Area'].map({'Yes' :1, 'No' :0}).astype('int')

In [131]:
dataset['Property_Area'] =dataset['Property_Area'].map({'Urban' :1,'Semiurban' :1, 'Rural' :0}).astype('int')

In [116]:
dataset['Loan_Status'] = dataset['Loan_Status'].map({'Y':1,'N':0}).astype('int')

In [122]:
dataset['Dependents'].unique()

array(['0', '1', '2', '4'], dtype=object)

In [132]:
dataset.sample(5)

Gender  Married Dependents  Education  Self_Employed  ApplicantIncome  \
127       1        0          0          1              0             3865   
381       1        0          0          1              0             5941   
439       0        1          0          0              1             7142   
37        0        1          0          1              0             3667   
465       1        0          0          1              0             6000   

     CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
127             1640.0       120.0             360.0             1.0   
381             4232.0       296.0             360.0             1.0   
439                0.0       138.0             360.0             1.0   
37              1459.0       144.0             360.0             1.0   
465                0.0       140.0             360.0             1.0   

     Property_Area  Loan_Status  
127              0            1  
381              1            1  
439              0            1  
37               1            1  
465              0            1

In [121]:
dataset['Dependents']=dataset['Dependents'].replace('3+', '4')

In [124]:
print(dataset.columns)

Index(['Gender', 'Married', 'Dependents', 'Education', 'Self_Employed',
       'ApplicantIncome', 'CoapplicantIncome', 'LoanAmount',
       'Loan_Amount_Term', 'Credit_History', 'Property_Area', 'Loan_Status'],
      dtype='object')


## Saving feature matrix to x and Target in Y

In [147]:
X=dataset.drop('Loan_Status', axis=1)

In [148]:
y=dataset['Loan_Status']

## Feature Scaling to handle large values in columns

In [143]:
dataset.sample(5)

Gender  Married Dependents  Education  Self_Employed  ApplicantIncome  \
9         1        1          1          1              0            12841   
474       1        0          2          1              0             5532   
605       1        1          0          0              0             2400   
456       1        1          0          1              0             4301   
521       1        0          0          1              0             2500   

     CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
9              10968.0       349.0             360.0             1.0   
474             4648.0       162.0             360.0             1.0   
605             3800.0       120.0             180.0             1.0   
456                0.0       118.0             360.0             1.0   
521                0.0        55.0             360.0             1.0   

     Property_Area  Loan_Status  
9                1            0  
474              0            1  
605              1            0  
456              1            1  
521              1            1

In [134]:
columns_feature_scaling=['ApplicantIncome','CoapplicantIncome','LoanAmount','Loan_Amount_Term']

In [149]:
from sklearn.preprocessing import StandardScaler
st = StandardScaler()
X[columns_feature_scaling]=st.fit_transform(X[columns_feature_scaling])

In [150]:
X

Gender  Married Dependents  Education  Self_Employed  ApplicantIncome  \
0         1        0          0          1              0         0.093956   
1         1        1          1          1              0        -0.122556   
2         1        1          0          1              1        -0.393281   
3         1        1          0          0              0        -0.464597   
4         1        0          0          1              0         0.119780   
..      ...      ...        ...        ...            ...              ...   
609       0        0          0          1              0        -0.410383   
610       1        1          4          1              0        -0.204132   
611       1        1          1          1              0         0.474135   
612       1        1          2          1              0         0.390506   
613       0        0          0          1              1        -0.122556   

     CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
0            -0.551193   -0.299444          0.277976             1.0   
1            -0.043947   -0.201796          0.277976             1.0   
2            -0.551193   -0.958572          0.277976             1.0   
3             0.241968   -0.299444          0.277976             1.0   
4            -0.551193   -0.043117          0.277976             1.0   
..                 ...         ...               ...             ...   
609          -0.551193   -0.897542          0.277976             1.0   
610          -0.551193   -1.275930         -2.514490             1.0   
611          -0.470464    1.323963          0.277976             1.0   
612          -0.551193    0.518362          0.277976             1.0   
613          -0.551193   -0.140766          0.277976             0.0   

     Property_Area  
0                1  
1                0  
2                1  
3                1  
4                1  
..             ...  
609              0  
610              0  
611              1  
612              1  
613              1  

[586 rows x 11 columns]

##  Splitting the data into train and test sets

In [146]:
from sklearn.model_selection import train_test_split
from sklearn.model_selection import cross_val_score
from sklearn.metrics import accuracy_score
import numpy as np

In [156]:
model_df={}
def model_val(model,X,y):
    X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2, random_state=35)
    
    model.fit(X_train,y_train)
    y_pred=model.predict(X_test)
    print(f"{model} accuracy is {accuracy_score(y_test,y_pred)}")
    
    score=cross_val_score(model,X,y,cv=5)
    print(f"{model} Avg cross val score is {np.mean(score)}")
    model_df[model]=round(np.mean(score)*100,2) 

In [157]:
model_df

{}

### Logistic Regression

Definition: Logistic Regression is a statistical method for modeling the probability of a binary outcome. It predicts the probability that an instance belongs to a particular category.

Use Case: Binary classification problems.

In [158]:
from sklearn.linear_model import LogisticRegression
model=LogisticRegression()
model_val(model,X,y)

LogisticRegression() accuracy is 0.847457627118644
LogisticRegression() Avg cross val score is 0.8054613935969869


### SVC (support vector classifier)

Definition: Support Vector Classifier (SVC) is a type of support vector machine (SVM) algorithm used for classification tasks. It finds a hyperplane that best separates data into different classes.

Use Case: Classification tasks where data points need to be separated by a hyperplane.

In [160]:
from sklearn import svm
model=svm.SVC()
model_val(model,X,y)

SVC() accuracy is 0.8135593220338984
SVC() Avg cross val score is 0.7969143850499782


In [161]:
model_df

{LogisticRegression(): 80.55, SVC(): 79.69}

### Decision Tree Classifier

Definition: Decision Tree is a tree-like model where each node represents a decision based on a feature, and each branch represents the outcome of the decision. It is used for both classification and regression tasks.

Use Case: Classification and regression problems with hierarchical decision-making.

In [162]:
from sklearn.tree import DecisionTreeClassifier
model = DecisionTreeClassifier()
model_val(model,X,y)

DecisionTreeClassifier() accuracy is 0.7033898305084746
DecisionTreeClassifier() Avg cross val score is 0.716673909894249


### Random Forest Classifier

Definition: Random Forest is an ensemble learning method that constructs a multitude of decision trees during training and outputs the mode (classification) or mean (regression) of the individual trees.

Use Case: Classification and regression tasks where multiple decision trees enhance predictive accuracy.

In [163]:
from sklearn.ensemble import RandomForestClassifier
model =RandomForestClassifier()
model_val(model,X,y)

RandomForestClassifier() accuracy is 0.7796610169491526
RandomForestClassifier() Avg cross val score is 0.7849630595393308


### Gradient Boosting Classifier

Definition: Gradient Boosting is an ensemble learning technique where weak models (usually decision trees) are added sequentially to correct errors made by the previous models. It builds a strong predictive model by combining the predictions of weak models.

Use Case: Classification and regression tasks where high predictive accuracy is required.


In [164]:
from sklearn.ensemble import GradientBoostingClassifier
model =GradientBoostingClassifier()
model_val(model,X,y)

GradientBoostingClassifier() accuracy is 0.788135593220339
GradientBoostingClassifier() Avg cross val score is 0.7883818629581342


##  HyperParameter tuning

In [165]:
from sklearn.model_selection import RandomizedSearchCV

### Logistic Regression

In [166]:
log_reg_grid={"C":np.logspace(-4,4,20),
             "solver":['liblinear']}

In [167]:
rs_log_reg=RandomizedSearchCV(LogisticRegression(),
                   param_distributions=log_reg_grid,
                  n_iter=20,cv=5,verbose=True)

In [168]:
rs_log_reg.fit(X,y)

Fitting 5 folds for each of 20 candidates, totalling 100 fits


RandomizedSearchCV(cv=5, estimator=LogisticRegression(), n_iter=20,
                   param_distributions={'C': array([1.00000000e-04, 2.63665090e-04, 6.95192796e-04, 1.83298071e-03,
       4.83293024e-03, 1.27427499e-02, 3.35981829e-02, 8.85866790e-02,
       2.33572147e-01, 6.15848211e-01, 1.62377674e+00, 4.28133240e+00,
       1.12883789e+01, 2.97635144e+01, 7.84759970e+01, 2.06913808e+02,
       5.45559478e+02, 1.43844989e+03, 3.79269019e+03, 1.00000000e+04]),
                                        'solver': ['liblinear']},
                   verbose=True)

In [169]:
rs_log_reg.best_score_

0.8054613935969869

In [170]:
rs_log_reg.best_params_

{'solver': 'liblinear', 'C': 0.23357214690901212}

### SVC

In [171]:
svc_grid = {'C':[0.25,0.50,0.75,1],"kernel":["linear"]}

In [172]:
rs_svc=RandomizedSearchCV(svm.SVC(),
                  param_distributions=svc_grid,
                   cv=5,
                   n_iter=20,
                  verbose=True)

In [173]:
rs_svc.fit(X,y)

C:\Users\user\anaconda3\envs\DAB200\lib\site-packages\sklearn\model_selection\_search.py:292: UserWarning: The total space of parameters 4 is smaller than n_iter=20. Running 4 iterations. For exhaustive searches, use GridSearchCV.
  warnings.warn(


Fitting 5 folds for each of 4 candidates, totalling 20 fits


RandomizedSearchCV(cv=5, estimator=SVC(), n_iter=20,
                   param_distributions={'C': [0.25, 0.5, 0.75, 1],
                                        'kernel': ['linear']},
                   verbose=True)

In [176]:
rs_svc.best_score_


0.8054613935969869

In [175]:
rs_svc.best_params_

{'kernel': 'linear', 'C': 0.25}

### Random Forest Classifier

In [177]:
RandomForestClassifier()

RandomForestClassifier()

In [178]:
rf_grid={'n_estimators':np.arange(10,1000,10),
  'max_features':['auto','sqrt'],
 'max_depth':[None,3,5,10,20,30],
 'min_samples_split':[2,5,20,50,100],
 'min_samples_leaf':[1,2,5,10]
 }

In [179]:
rs_rf=RandomizedSearchCV(RandomForestClassifier(),
                  param_distributions=rf_grid,
                   cv=5,
                   n_iter=20,
                  verbose=True)

In [180]:
rs_rf.fit(X,y)

Fitting 5 folds for each of 20 candidates, totalling 100 fits


RandomizedSearchCV(cv=5, estimator=RandomForestClassifier(), n_iter=20,
                   param_distributions={'max_depth': [None, 3, 5, 10, 20, 30],
                                        'max_features': ['auto', 'sqrt'],
                                        'min_samples_leaf': [1, 2, 5, 10],
                                        'min_samples_split': [2, 5, 20, 50,
                                                              100],
                                        'n_estimators': array([ 10,  20,  30,  40,  50,  60,  70,  80,  90, 100, 110, 120, 130,
       140, 150, 160, 170, 180, 190, 200, 210, 220, 230, 240, 250, 260,
       270, 280, 290, 300, 310, 320, 330, 340, 350, 360, 370, 380, 390,
       400, 410, 420, 430, 440, 450, 460, 470, 480, 490, 500, 510, 520,
       530, 540, 550, 560, 570, 580, 590, 600, 610, 620, 630, 640, 650,
       660, 670, 680, 690, 700, 710, 720, 730, 740, 750, 760, 770, 780,
       790, 800, 810, 820, 830, 840, 850, 860, 870, 880, 890, 900, 9

In [181]:
rs_rf.best_score_

0.8054613935969869

In [182]:
rs_rf.best_params_

{'n_estimators': 450,
 'min_samples_split': 100,
 'min_samples_leaf': 1,
 'max_features': 'auto',
 'max_depth': 20}

 ### Comparision
 
 LogisticRegression score Before Hyperparameter Tuning: 80.54
 
LogisticRegression score after Hyperparameter Tuning: 80.54 
    
------------------------------------------------------
SVC score Before Hyperparameter Tuning: 79.69

SVC score after Hyperparameter Tuning: 80.54
    
--------------------------------------------------------
RandomForestClassifier score Before Hyperparameter Tuning: 78.49

RandomForestClassifier score after Hyperparameter Tuning: 80.54 

### Saving the Model with best parameter

In [185]:
X = dataset.drop('Loan_Status',axis=1)
y = dataset['Loan_Status']

In [186]:
rf = RandomForestClassifier(n_estimators=270,
 min_samples_split=5,
 min_samples_leaf=5,
 max_features='sqrt',
 max_depth=5)

In [187]:
rf.fit(X,y)

RandomForestClassifier(max_depth=5, max_features='sqrt', min_samples_leaf=5,
                       min_samples_split=5, n_estimators=270)

In [188]:
import joblib

In [189]:
joblib.dump(rf,'loan_status_predict')

['loan_status_predict']

In [190]:
model = joblib.load('loan_status_predict')

In [198]:
import pandas as pd
df = pd.DataFrame({
    'Gender':1,
    'Married':1,
    'Dependents':2,
    'Education':0,
    'Self_Employed':1,
    'ApplicantIncome':4261,
    'CoapplicantIncome':0.0,
    'LoanAmount':146,
    'Loan_Amount_Term':360,
    'Credit_History':1,
    'Property_Area':1
},index=[0])

In [202]:
import pandas as pd
df1 = pd.DataFrame({
    'Gender':1,
    'Married':1,
    'Dependents':2,
    'Education':0,
    'Self_Employed':0,
    'ApplicantIncome':2889,
    'CoapplicantIncome':0.0,
    'LoanAmount':45,
    'Loan_Amount_Term':180,
    'Credit_History':0,
    'Property_Area':1
},index=[0])

In [203]:
df1

Gender  Married  Dependents  Education  Self_Employed  ApplicantIncome  \
0       1        1           2          0              0             2889   

   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
0                0.0          45               180               0   

   Property_Area  
0              1

In [197]:
X.head(5)

Gender  Married Dependents  Education  Self_Employed  ApplicantIncome  \
0       1        0          0          1              0             5849   
1       1        1          1          1              0             4583   
2       1        1          0          1              1             3000   
3       1        1          0          0              0             2583   
4       1        0          0          1              0             6000   

   CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
0                0.0       120.0             360.0             1.0   
1             1508.0       128.0             360.0             1.0   
2                0.0        66.0             360.0             1.0   
3             2358.0       120.0             360.0             1.0   
4                0.0       141.0             360.0             1.0   

   Property_Area  
0              1  
1              0  
2              1  
3              1  
4              1

In [209]:
result = model.predict(df)

In [208]:
if result1==1:
    print("Loan Approved")
else:
    print("Loan Not Approved")

Loan Not Approved


In [210]:
if result==1:
    print("Loan Approved")
else:
    print("Loan Not Approved")

Loan Approved
